### PR TITLE
Enable RTC again in D-Lab hub

### PR DESCRIPTION
**Pull request by balajialg**

### CODE REVIEW

**Comment by balajialg** [comment]
Ref #4223

**Comment by aculich** [comment]
Thanks! As soon as this is ready let me know and we'll test it out!

**Comment by balajialg** [comment]
@aculich Looks like we have some complications with enabling RTC in D-lab hub. We need to upgrade the following packages to make RTC work in the D-Lab hub. 

```
- altair==4.2.2
  - boken==2.4.3
  - dask==2023.1.1
  - jupyter_server==2.2.1
  - jupyterlab==3.6.1
  - jupyterlab_server==2.19.0
  - tornado==6.2.0
  - git+https://github.com/berkeley-dsep-infra/tmpystore.git@84765e1
```
However, the D-Lab hub shares the image with the main Datahub. These upgrades are unsafe and may impact instructors who use the main Datahub. The other option is to use Stat 159 hub's image so that you can get these packages along with the corresponding versions. However, Stat 159 image is primarily python specific. Do you require R packages for the D-Lab hub? I am assuming yes by looking at the workshop list [here](https://dlab.berkeley.edu/training/upcoming-workshops). Please do clarify.

If you require R packages then we need to provision a new image for D-Lab hub which might take sometime as there aren't enough cycles this sprint.

**Comment by aculich** [comment]
@balajialg Yes, we do need to continue having R support. I'm wondering if maybe we should create a dev/testing instance alongside our current one?

Maybe: dlab-testing.datahub.berkeley.edu
Alongside: dlab.datahub.berkeley.edu

So that way we can easily test out breaking changes like this before deploying to our main hub?

Maybe we could even set up a [blue green deployment](https://www.redhat.com/en/topics/devops/what-is-blue-green-deployment), since I think we'll be doing this kind of bleeding edge testing more and more in the future with other features or images.

In any case, we don't want to break existing compatibility for our workshops happening right now. So a separate testing environment would be handy— and we might even pilot a workshop or two using it at some point sooner.

Most of the github repos that we're using come from one place: https://github.com/dlab-berkeley/
Which are the github repos that the [upcoming workshop list](https://dlab.berkeley.edu/training/upcoming-workshops) points to.

**Comment by aculich** [comment]
Also, if we do need to switch over the existing hub, we have a window of time from  Thu March 16 through Friday March 24th to do it because we have no workshop happening between those dates (starting back up again 4pm March 27th for an R workshop).

[D-Lab Events Calendar](https://dlab.berkeley.edu/events/calendar)
image

**Comment by balajialg** [comment]
@aculich Thanks for sharing the detailed requirements! I created an issue for to explore this requirement in detail (#4375) and evaluate whether creating a new hub is the way forward. You can track further updates about this requirement there.

I added the Stat 159 image to D-Lab hub as part of my latest commit. Would it be possible for anyone from your team to add/install the required R packages and RStudio to this image? This image already has all the RTC related changes. That will make it easy at our end. Please let me know if that's feasible.

File: deployments/dlab/image/environment.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+name: stat159-s23
```
Comment (shaneknapp) [inline deployments/dlab/image/environment.yml:1]: should probably change this...

**Review by shaneknapp — COMMENTED** [review]
other than my previous comment, lgtm

**Comment by balajialg** [comment]
> I added the Stat 159 image to D-Lab hub as part of my latest commit. Would it be possible for anyone from your team to add/install the required R packages and RStudio to this image? This image already has all the RTC related changes. That will make it easy at our end. Please let me know if that's feasible.

@aculich Any updates on this particular request? RTC changes are currently enabled as part of this PR and only the R-related changes to the image (Adding R packages + RStudio) are pending. Does anyone from your team have the bandwidth to help with this requirement?

**Comment by aculich** [comment]
@psachdeva can you try testing it out and let Balaji know if you have any issues or requests for packages updates?

**Comment by aculich** [comment]
@balajialg how would @psachdeva test this out?

I see this PR says "Merging is blocked" and Pratik says he doesn't see it active in our hub.

image

**Comment by pssachdeva** [comment]
Happy to hop on a huddle if we need to discuss further.

**Comment by balajialg** [comment]
@pssachdeva @aculich I have set up a meeting on Friday to talk through this (next available slot I could find in bcal). Merging rights are currently available only for hub admins and that is the reason why it appears "Merging is blocked". We can talk during this meeting about providing merge access in the Datahub repo. The next steps for this PR are,

a) Make image changes as a PR  that includes R packages and RStudio in the D-Lab image
b) Merge it to staging and iteratively test a few notebooks to see whether they are working
c) Merge it to prod during a time frame where there are no scheduled workshops so that this is not a disruptive action for D-Lab users

**Comment by balajialg** [comment]
@aculich @pssachdeva Closing this PR in favor of #4484

**Pull request closed without merging**